python/lsst/rubintv/production/starTracker.py

```diff
@@ -479,3 +511,128 @@ def callback(self, filename):
+
+    Parameters
+    ----------
+    locationConfig : `lsst.rubintv.production.utils.LocationConfig`
+        The locationConfig containing the path configs.
+    dayObs : `int`, optional
+        TODO: XXX document this!!! XXX
+    embargo : `bool`, optional
+        Use the embargo repo?
+    doRaise : `bool`, optional
+        If True, raise exceptions instead of logging them as warnings.
+    """
+
+    def __init__(self, locationConfig, *, dayObs=None, doRaise=False):
+        name = 'starTrackerNightReport'
+        log = logging.getLogger(f'lsst.rubintv.production.{name}')
+        self.rootDataPath = locationConfig.starTrackerDataPath
+
+        # we have to pick a camera to watch for data on for now, and while we
+        # are always reading all three out together it doesn't matter which we
+        # pick, so pick the regular one for now (but don't choose fast as that
+        # is also a DIMM and so is more likely to be relocated). This won't be
+        # a problem once we move to ingesting data and using the butler, so the
+        # temp/hacky nature of this is fine for now.
+        watcher = StarTrackerWatcher(rootDataPath=self.rootDataPath,
+                                     bucketName=locationConfig.bucketName,
+                                     camera=regularCam)  # tie the night report to the regular cam for now
+
+        super().__init__(locationConfig=locationConfig,
+                         log=log,
+                         watcher=watcher,
+                         doRaise=doRaise
+                         )
+
+        self.dayObs = dayObs if dayObs else getCurrentDayObs_int()
+
+    def getMetadataTableContents(self):
+        """Get the measured data for the current night.
+
+        Returns
+        -------
+        mdTable : `pandas.DataFrame`
+            The contents of the metdata table from the front end.
+        """
+        sidecarFilename = os.path.join(self.locationConfig.starTrackerMetadataPath,
+                                       f'dayObs_{self.dayObs}.json')
+
+        try:
+            mdTable = pd.read_json(sidecarFilename).T
+            mdTable = mdTable.sort_index()
+        except Exception as e:
+            self.log.warning(f"Failed to load metadata table from {sidecarFilename}: {e}")
+            return None
+
+        if mdTable.empty:
+            return None
+
+        return mdTable
+
+    def createPlotsAndUpload(self):
+        """Create and upload all plots defined in nightReportPlots.
+
+        All plots defined in __all__ in nightReportPlots are discovered,
+        created and uploaded. If any fail, the exception is logged and the next
+        plot is created and uploaded.
+        """
+        md = self.getMetadataTableContents()
+        self.log.info(f'Creating plots for dayObs {self.dayObs} with: '
+                      f'{0 if md is None else len(md)} items in the metadata table')
+
+        for plotClassName in starTrackerNightReportPlots.__all__:
```

Comment:
Having a dedicated module-scope variable for the plots to include here is probably better than `__all__`, even if you set one from the other in the module.  Even better would be a function in that module that could be called to create all of the plots - and that might provide the solution to the e1 vs e2 duplication I commented on elsewhere.

Same argument applies for LATISS plots, of course.